content/en/blog/_posts/2024-04-xx-structured-authz-beta.md

```diff
@@ -0,0 +1,324 @@
+        #      - If failurePolicy=NoOpinion, then the error is ignored and the webhook is skipped
+      matchConditions:
+      # expression represents the expression which will be evaluated by CEL. Must evaluate to bool.
+      # CEL expressions have access to the contents of the SubjectAccessReview in v1 version.
+      # If version specified by subjectAccessReviewVersion in the request variable is v1beta1,
+      # the contents would be converted to the v1 version before evaluating the CEL expression.
+      #
+      # Documentation on CEL: https://kubernetes.io/docs/reference/using-api/cel/
+      #
+      # only send resource requests to the webhook
+      - expression: has(request.resourceAttributes)
+      # only intercept requests to kube-system
+      - expression: request.resourceAttributes.namespace == 'kube-system'
+      # don't intercept requests from kube-system service accounts
+      - expression: !('system:serviceaccounts:kube-system' in request.user.groups)
+  - type: Node
+    name: node
+  - type: RBAC
+    name: rbac
+  - type: Webhook
+    name: in-cluster-authorizer
+    webhook:
+      authorizedTTL: 5m
+      unauthorizedTTL: 30s
+      timeout: 3s
+      subjectAccessReviewVersion: v1
+      failurePolicy: NoOpinion
+      connectionInfo:
+        type: InClusterConfig
+```
+
+The following configuration examples illustrate real world scenarios that need
+the ability to specify multiple webhooks with distinct settings, precedence
+order, and failure modes.
+
+### Protecting Installed CRDs
+Ensuring the availability of Custom Resource Definitions (CRDs) at cluster
+startup has been a key demand, One of the blockers for having a controller
+reconciling those CRDs is to have a protection mechanism for them, which can be
+achieved through multiple Authorization Webhooks. This was not possible before.
+Now with the Structured Authorization Configuration feature, administrators can
+specify multiple webhooks offering a solution where RBAC falls short, especially
+in denying permissions to 'non-system' users for certain CRDs.
+
+Assuming the following for this scenario:
+- The "protected" CRDs are installed in the kube-system namespace.
+- They can only be modified by users in the group
+  `system:serviceaccounts:kube-superuser`
+
+```yaml
+apiVersion: apiserver.config.k8s.io/v1beta1
+kind: AuthorizationConfiguration
+authorizers:
+  - type: Webhook
+    name: system-crd-protector
+    webhook:
+      unauthorizedTTL: 30s
+      timeout: 3s
+      subjectAccessReviewVersion: v1
+      matchConditionSubjectAccessReviewVersion: v1
+      failurePolicy: Deny
+      connectionInfo:
+        type: KubeConfig
+        kubeConfigFile: /files/kube-system-authz-webhook.yaml
+      matchConditions:
+      # only send resource requests to the webhook
+      - expression: has(request.resourceAttributes)
+      # only intercept requests to kube-system
+      - expression: request.resourceAttributes.namespace == 'kube-system'
```

Comment:
CRDs are namespaced, so this condition means this webhook wouldn't intercept any CRD requests ... should this be 
```
- expression: request.resourceAttributes.resource.resource = "customresourcedefinitions"
- expression: request.resourceAttributes.resource.group = ""
```

Reply:
Do you mean the CRDs (not CRs) are cluster scoped and therefore, we should remove this check 
>  expression: request.resourceAttributes.namespace == 'kube-system'

with

>  expression: request.resourceAttributes.resource.resource = "customresourcedefinitions" 
>  expression: request.resourceAttributes.resource.group = ""

Reply:
oops, yes 🤦